data/scripts.ts

```diff
@@ -335,7 +335,7 @@ export const Scripts: BattleScriptsData = {
 			[moveSteps[2], moveSteps[4]] = [moveSteps[4], moveSteps[2]];
 		}
 
-		this.setActiveMove(move, pokemon, targets[0]);
+		if (!notActive) this.setActiveMove(move, pokemon, targets[0]);
```

Comment:
Maybe this really belongs in `useMoveInner` somewhere? (That way you wouldn't need it in `tryMoveHit` either.)

Reply:
The futuremove condition calls trySpreadMoveHit directly. Is it advisable to change this?

Reply:
(I'm insufficiently familiar with what effects can modify future moves to be able to contribute meaningfully to that question.)